pyodide-build/pyodide_build/buildpkg.py

```diff
@@ -547,6 +550,16 @@ def package_wheel(
         finally:
             shutil.rmtree(test_dir, ignore_errors=True)
 
+    if extract_metadata_file:
+        with TemporaryDirectory() as temp_dir:
+            unpack_wheel(wheel, Path(temp_dir))
```

Comment:
There is no need to unzip the whole thing. You can use [zipfile](https://docs.python.org/3/library/zipfile.html) to only load the one relevant file. A wheel is  zip file.